### PR TITLE
New rule: `at-if-no-null`

### PR DESCRIPTION
**Pull request by rambleraptor**
The [sass/linter](https://github.com/sass/linter/blob/master/lib/src/rules/use_falsey_null.dart) project had a rule to check for using `== null` or `!= null` inside @if statements.

This rule replicates that.

### CODE REVIEW

**Comment by rambleraptor** [comment]
@nex3

File: src/rules/at-if-no-null/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }`,
+      description: "does not use the != null format"
+    },
+    {
+      code: `a {
+      @if not ($x) {}
+    }`,
+      description: "does not use the == null format"
+    }
+  ],
```
Comment (kristerkari) [inline src/rules/at-if-no-null/__tests__/index.js]: There could be some more tests with calculations and other conditions to make sure that there are no false positives/negatives.

File: src/rules/at-if-no-null/index.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { namespace } from "../../utils";
+import { utils } from "stylelint";
+
+export const ruleName = namespace("at-if-no-null");
+
+export const messages = utils.ruleMessages(ruleName, {
+  equals_null: "Expected @if(statement) rather than @if(statement == null)",
+  not_equals_null: "Expected @if(!statement) rather than @if(statement != null)"
```
Comment (kristerkari) [inline src/rules/at-if-no-null/index.js]: Are `@if not ($x)` and `@if(!$x)` both valid? Maybe the error messages and tests should have both examples?

File: src/rules/at-if-no-null/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# at-if-no-null
+
+Check for equality to null is unnecessarily explicit since `null` is falsey in Sass.
+
+```scss
+a {
+    @if ($x == null) {}
```
Comment (nex3) [inline src/rules/at-if-no-null/README.md]: I'd like to discourage the use of parentheses with `@if` as well, so maybe these examples should omit them too?

File: src/rules/at-if-no-null/__tests__/index.js (hunk 1 above)
Comment (rambleraptor) [reply src/rules/at-if-no-null/__tests__/index.js]: @nex3 any ideas on good tests or language constructs that users might be using in these if/else statements?

File: src/rules/at-if-no-null/index.js (hunk 2 above)
Comment (rambleraptor) [reply src/rules/at-if-no-null/index.js]: I changed the messages a little bit. How do they sound now?

File: src/rules/at-if-no-null/__tests__/index.js (hunk 1 above)
Comment (nex3) [reply src/rules/at-if-no-null/__tests__/index.js]: When I'm looking for examples, I often look through big open-source corpuses like https://github.com/angular/components/tree/master/src/material or https://github.com/carbon-design-system/carbon/tree/master/packages.

File: src/rules/at-if-no-null/index.js (hunk 2 above)
Comment (kristerkari) [reply src/rules/at-if-no-null/index.js]: Sounds good now.

File: src/rules/at-if-no-null/__tests__/index.js (hunk 1 above)
Comment (rambleraptor) [reply src/rules/at-if-no-null/__tests__/index.js]: Great idea! I didn't know of any large Sass codebases.

I grepped through carbon-design and came across a common pattern (if $x != null and expr). I made adjustments to the rule for that pattern.

**Pull request merged**